app/views/users/_admin_nav.html.erb

```diff
@@ -0,0 +1,10 @@
+<% if current_user && current_user.admin? %>
```

Comment:
Had to add a check that current_user is not nil before checking if current_user is an admin, merchant, or user. Seems like there's even more room for refactoring this but I'm not sure what that would be

Reply:
Maybe your `current_admin?` `current_merchant?` methods.